mergin/utils.py

```diff
@@ -145,3 +146,83 @@ def get_versions_with_file_changes(
             idx_to = idx
             break
     return [f"v{ver_nr}" for ver_nr in all_version_numbers[idx_from:idx_to + 1]]
+
+
+def unique_file_name(path):
```

Comment:
if the function works for both files and directories, maybe better to name it e.g. `unique_path_name()` ?